message_app/send_message.py

```diff
@@ -98,18 +102,58 @@ def message_handler(data):
     message = data["message"]
     conversation_id = data["conversation_id"]
     created_at = data["created_at"]
+    attachment_name = data["attachment_name"]
 
     # Update the database
     user_id = User.select(username).id
     conversation = Conversations.select(conversation_id)
-    new_message = Messages(sender_id=user_id, content=message, created_at=created_at, conversation_id=conversation_id)
+    new_message = Messages(sender_id=user_id, content=message, created_at=created_at, conversation_id=conversation_id, attachment_name=attachment_name)
     Messages.insert(new_message)
     conversation.last_message_id = new_message.id # Manually set the last_message_id for now
     DB.session.commit()
     data["id"] = new_message.id
 
     emit("message_handler_client", data, room=conversation_id)
 
+# A socket that handles sending/receiving images
+@socketio.on("image_handler_server", namespace="/messages")
+def image_handler(data):
+    # The data is a dictionary with five keys:
+    # "sender_name": The username of the sender
+    # "conversation_id": The current conversation's id
+    # "image_name": The name of the image
+    # "image": The image as a binary file.
+    # "timestamp": The time that user sent this image
+
+    sender_name = data["sender_name"] # This variable will be used in the future to handle duplicated name issue
```

Comment:
why is it an issue?

Reply:
For now, every time the server receives an image with the same name, the old image will be deleted. I'm planning to concatenate the sender_name and the timestamp to the image's name to avoid duplication.

Reply:
You can skip them for now. We will have to design our media storage properly.